src/lib/util/time-ranges.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  TimeRangeName,
+  TimeSeriesTimeRange,
+} from "$common/database-service/DatabaseTimeSeriesActions";
+
+const makeTimeRange = (
+  name: TimeRangeName,
+  end: Date = new Date()
+): TimeSeriesTimeRange => {
+  switch (name) {
+    case TimeRangeName.LastHour:
```

Comment:
Lot of common code here. Perhaps we can replace with a map that pulls start and end date? It wont probably handle all use cases. We can have dedicated case statement for those.